moped-editor/src/views/projects/projectView/ProjectFundingTable.js

```diff
@@ -324,96 +382,84 @@ const ProjectFundingTable = () => {
-                        {error.message}
-                      </span>
-                    ),
-                    severity: "error",
-                  });
+            addProjectFunding({
+              variables: {
+                objects: {
+                  ...newData,
+                  project_id: projectId,
+                  added_by: getDatabaseId(user),
+                  funding_status_id: 1,
+                },
+              },
+            })
+              .then(() => refetch())
+              .catch(error => {
+                setSnackbarState({
+                  open: true,
+                  message: (
+                    <span>
+                      There was a problem adding funding. Error message:{" "}
+                      {error.message}
+                    </span>
+                  ),
+                  severity: "error",
                 });
+              }),
+          onRowUpdate: (newData, oldData) => {
+            const updateProjectFundingData = newData;
 
-                setTimeout(() => refetch(), 501);
-                resolve();
-              }, 500);
-            }),
-          onRowUpdate: (newData, oldData) =>
-            new Promise((resolve, reject) => {
-              setTimeout(() => {
-                let updateProjectFundingData = newData;
-
-                // Remove unexpected variables
-                delete updateProjectFundingData.__typename;
-                delete updateProjectFundingData.added_by;
-                delete updateProjectFundingData.date_added;
-                // Format edited funding values to number value
-                updateProjectFundingData.funding_amount = Number(
-                  newData.funding_amount
-                );
-                // add fallback of empty strings instead of null value
-                updateProjectFundingData.fund_dept_unit =
-                  newData.fund_dept_unit || "";
-                updateProjectFundingData.funding_description =
-                  newData.funding_description || "";
-                updateProjectFundingData.funding_program_id =
-                  newData.funding_program_id || 0;
+            // Remove unexpected variables
+            delete updateProjectFundingData.__typename;
+            delete updateProjectFundingData.added_by;
+            delete updateProjectFundingData.date_added;
+            // Format edited funding values to number value
+            updateProjectFundingData.funding_amount = Number(
+              newData.funding_amount
+            );
+            // add fallback of empty strings instead of null value
+            updateProjectFundingData.fund_dept_unit =
+              newData.fund_dept_unit || "";
+            updateProjectFundingData.funding_description =
+              newData.funding_description || "";
+            updateProjectFundingData.funding_program_id =
+              newData.funding_program_id || 0;
```

Comment:
It's tedious that we have to do all this data cleaning but I don't know of a better solution. Just feels like a lot of extra boilerplate.